pkg/meta/sql.go

```diff
@@ -1336,44 +1342,21 @@ func (m *dbMeta) Readdir(ctx Context, inode Ino, plus uint8, entries *[]*Entry)
 		Attr:  &pattr,
 	})
 
-	rows, err := m.engine.Rows(&edge{Parent: inode})
-	if err != nil {
-		return errno(err)
-	}
-	names := make(map[Ino][]byte)
-	var inodes []string
-	for rows.Next() {
-		var e edge
-		err = rows.Scan(&e)
-		if err != nil {
-			_ = rows.Close()
-			return errno(err)
-		}
-		names[e.Inode] = []byte(e.Name)
-		inodes = append(inodes, strconv.FormatUint(uint64(e.Inode), 10))
+	dbSession := m.engine.Table(&edge{})
+	if plus == 1 {
+		dbSession = dbSession.Table(&node{}).Join("INNER", "jfs_edge", "jfs_edge.inode=jfs_node.inode")
 	}
-	_ = rows.Close()
-	if len(inodes) == 0 {
-		return 0
-	}
-
-	var n node
-	nodes, err := m.engine.Where(fmt.Sprintf("inode IN (%s)", strings.Join(inodes, ","))).Rows(&n)
-	if err != nil {
+	var nodesWithParent []nodeParent
+	if err := dbSession.Find(&nodesWithParent, &edge{Parent: inode}); err != nil {
 		return errno(err)
 	}
-	defer nodes.Close()
-	for nodes.Next() {
-		err = nodes.Scan(&n)
-		if err != nil {
-			return errno(err)
-		}
-		attr := new(Attr)
-		m.parseAttr(&n, attr)
+	for _, n := range nodesWithParent {
+		var attr Attr
+		m.parseAttr(&n.node, &attr) // .Type is fetched into node, so attr will always have .Typ filled regardless of plus value
```

Comment:
parseAttr() will always have `attr.Full = true`, but when `plus==0`, we should have a `attr.Full = false`, and only Typ in attr.

Reply:
OK, will fix it. I have no idea how `attr.Full` is used.